lib/classes/Variables.test.js

```diff
@@ -123,6 +157,44 @@ describe('Variables', () => {
       });
     });
 
+    it('should call populateObject if variable value is an object', () => {
+      const serverless = new Serverless();
+      serverless.variables.options = {
+        stage: 'prod',
+      };
+      const property = '${opt:stage}';
+      const variableValue = {
+        stage: '${opt:stage}',
+      };
+      const variableValuePopulated = {
+        stage: 'prod',
+      };
+
+      serverless.variables.loadVariableSyntax();
+
+      const populateObjectStub = sinon
+        .stub(serverless.variables, 'populateObject')
+        .resolves(variableValuePopulated);
+      const getValueFromSourceStub = sinon
+        .stub(serverless.variables, 'getValueFromSource')
+        .resolves(variableValue);
+      const populateVariableStub = sinon
+        .stub(serverless.variables, 'populateVariable')
+        .resolves(variableValuePopulated);
+
+      return serverless.variables.populateProperty(property).then(newProperty => {
+        expect(populateObjectStub.called).to.equal(true);
+        expect(getValueFromSourceStub.called).to.equal(true);
+        expect(populateVariableStub.called).to.equal(true);
+        expect(newProperty).to.deep.equal(variableValuePopulated);
+
+        serverless.variables.populateObject.restore();
+        serverless.variables.getValueFromSource.restore();
+        serverless.variables.populateVariable.restore();
+        return Promise.resolve();
```

Comment:
Could we stick to one Promise implementation from now on (even in tests)?

I'll fix this one in the Async follow-up PR